### ISSUE TITLE
Support config files without extension

### ISSUE THREAD

**Issue by cpheps** [spf13/viper#501]
It would be nice if a config file does not need an extension. It's not uncommon for a dot file to not have an extension, but be in a metadata format. 

In order to solve how to parse the file would be ideal to use `viper.SetConfigType()`.

Example:

Reading in a config file `$HOME/.config` that is YAML formatted.

```
viper.AddConfigPath(homeDir)
viper.SetConfigType("yaml")
viper.SetConfigName(".config")
```

**Comment by zillani-nuovo** [comment]
Any update on this, We are waiting for this feature too!

**Comment by chrim5** [comment]
+1 !

**Comment by robertocromero** [comment]
We're waiting for this too.

**Comment by sagikazarmark** [comment]
This has been implemented in #722
